[WebXR] Adjust XRSession visible test

djg opened the pull request:
#### f67af466054c043a299acea2fefb5ae0da57a534
<pre>
[WebXR] Adjust XRSession visible test
<a href="https://bugs.webkit.org/show_bug.cgi?id=256938">https://bugs.webkit.org/show_bug.cgi?id=256938</a>
<a href="https://rdar.apple.com/problem/109489889">rdar://problem/109489889</a>

Reviewed by Dean Jackson.

XRVisibilityState is tristate: Visible, VisibleBlurred, Hidden. Previously only
Visible was considered &quot;visible&quot; causing a JS SecurityError to be thrown when
reporting poses in the &quot;blurred&quot; state, ie. when WebXR doesn&apos;t have input but
should render.

This change considers both Visible states as &quot;visible&quot; and Hidden as &quot;not
visible&quot;.

* Source/WebCore/Modules/webxr/WebXRSession.cpp:
(WebCore::WebXRSession::posesCanBeReported const):

Canonical link: <a href="https://commits.webkit.org/273337@main">https://commits.webkit.org/273337@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/aebf7ff301740d1b6491c627bc7a71d2ad68a50e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/35104 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/14033 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/37226 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/37864 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/31702 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/36265 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/16412 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/11103 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/30628 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/35649 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/11881 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/31319 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/10408 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/10475 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/31424 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/39111 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/31945 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/31753 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/36482 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/10584 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/8513 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/34475 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/12368 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8044 "Built successfully and passed tests") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/11112 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/11430 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

svillar reviewed (commented):
Makes sense but doesn't it make any new WPT test pass?

grorg reviewed (approved):
Yeah - if we don't have a WPT maybe we should make one?

djg committed:
[WebXR] Adjust XRSession visible test
https://bugs.webkit.org/show_bug.cgi?id=256938
rdar://problem/109489889

Reviewed by Dean Jackson.

XRVisibilityState is tristate: Visible, VisibleBlurred, Hidden. Previously only
Visible was considered "visible" causing a JS SecurityError to be thrown when
reporting poses in the "blurred" state, ie. when WebXR doesn't have input but
should render.

This change considers both Visible states as "visible" and Hidden as "not
visible".

* Source/WebCore/Modules/webxr/WebXRSession.cpp:
(WebCore::WebXRSession::posesCanBeReported const):

Canonical link: https://commits.webkit.org/273337@main
